src/NHibernate/Criterion/SimpleProjection.cs

```diff
@@ -26,11 +26,18 @@ public virtual IType[] GetTypes(string alias, ICriteria criteria, ICriteriaQuery
 			return null;
 		}
 
+		// Since v5.3
+		[Obsolete("This method has no more usage in NHibernate and will be removed in a future version.")]
```

Comment:
Not sure why it's been left. But it doesn't work properly for multi-column projections. Maybe to allow to retrieve default alias name outside of criteria context? So let's expose instead `GetColumnAlias(int position, int columnIndex)`